metrics_sdk/lib/opentelemetry/sdk/metrics/export/in_memory_metric_pull_exporter.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Metrics
+      module Export
+        class InMemoryMetricPullExporter < MetricReader
+          PREFERRED_TEMPORALITY = 'delta'
+
+          attr_reader :metric_snapshots, :metric_store
+
+          def initialize
+            @metric_snapshots = []
+            @mutex = Mutex.new
+            @metric_store = OpenTelemetry::SDK::Metrics::State::MetricStore.new
```

Comment:
The superclass also initializes this instance variable, but you're not calling `super` here.